lib/driver.js

```diff
@@ -264,8 +264,13 @@ class XCUITestDriver extends BaseDriver {
     this.logEvent('appConfigured');
 
     // fail very early if the app doesn't actually exist
+    // or if bundle id doesn't point to an installed app
     if (this.opts.app) {
       await checkAppPresent(this.opts.app);
+    } else if (this.opts.bundleId && !this.safari) {
```

Comment:
is there a reason to have a nested conditional rather than to put all the conditions in the `else if`?